# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/x86_64-getpriority.patch
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

On x86-64 the getpriority value returned by the kernel is biased by 20
to avoid negative return values.

  - Rene Rebe <rene@exactcode.de>

--- /dev/null	2007-01-29 09:48:37.372192192 +0100
+++ dietlibc/x86_64/getpriority.S	2007-01-30 14:23:16.001056730 +0100
@@ -0,0 +1,12 @@
+#include "syscalls.h"
+
+.text
+.global getpriority
+.type   getpriority,@function
+getpriority:
+	mov	$__NR_getpriority,%al
+	xorq	%rsi,%rsi
+	call	__unified_syscall
+	neg	%rax	/* the value is biased by 20 to avoid negative values */
+	add 	$20,%rax
+	ret
